pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java

```diff
@@ -287,8 +287,8 @@ private void handleTransactionMessage(TxnID txnId, Position position) {
                 .checkAbortedTransaction(txnId)) {
             ongoingTxns.put(txnId, (PositionImpl) position);
             PositionImpl firstPosition = ongoingTxns.get(ongoingTxns.firstKey());
-            //max read position is less than first ongoing transaction message position, so entryId -1
-            maxReadPosition = PositionImpl.get(firstPosition.getLedgerId(), firstPosition.getEntryId() - 1);
+            // max read position is less than first ongoing transaction message position
+            maxReadPosition = getPreviousPosition(firstPosition);
```

Comment:
In fact, `entryId=-1` does not cause program exceptions, There are many uses of this in pulsar. But if modified here, it is also acceptable

Reply:
There are two places that we need to get the previous position, but they are not consistent. We have better make them consistent.

Reply:
it makes sense